src/main/java/com/powsybl/openloadflow/ac/VoltageMagnitudeInitializer.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.ac;
+
+import com.google.common.base.Stopwatch;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.math.matrix.MatrixFactory;
+import com.powsybl.openloadflow.equations.*;
+import com.powsybl.openloadflow.network.*;
+import com.powsybl.openloadflow.network.util.VoltageInitializer;
+import gnu.trove.list.array.TDoubleArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This voltage initializer is able to find a voltage magnitude starting point by resolving a linear system
+ * using only voltage set points, branches reactance and branches voltage ratio.
+ * This initializer is particularly useful for cases with a large range of voltage (many transformers with a ratio far
+ * from 1pu for instance).
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class VoltageMagnitudeInitializer implements VoltageInitializer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VoltageMagnitudeInitializer.class);
+
+    public enum InitVmEquationType implements Quantity {
+        BUS_TARGET_V("v", ElementType.BUS),
+        BUS_ZERO("bus_z", ElementType.BUS);
+
+        private final String symbol;
+
+        private final ElementType elementType;
+
+        InitVmEquationType(String symbol, ElementType elementType) {
+            this.symbol = symbol;
+            this.elementType = elementType;
+        }
+
+        @Override
+        public String getSymbol() {
+            return symbol;
+        }
+
+        @Override
+        public ElementType getElementType() {
+            return elementType;
+        }
+    }
+
+    public enum InitVmVariableType implements Quantity {
+        BUS_V("v", ElementType.BUS);
+
+        private final String symbol;
+
+        private final ElementType elementType;
+
+        InitVmVariableType(String symbol, ElementType elementType) {
+            this.symbol = symbol;
+            this.elementType = elementType;
+        }
+
+        @Override
+        public String getSymbol() {
+            return symbol;
+        }
+
+        @Override
+        public ElementType getElementType() {
+            return elementType;
+        }
+    }
+
+    public static final class InitVmBusEquationTerm extends AbstractBusEquationTerm<InitVmVariableType, InitVmEquationType> {
+
+        private final List<Variable<InitVmVariableType>> variables;
+
+        private final TDoubleArrayList der;
+
+        private static boolean isConnected(LfBranch branch) {
```

Comment:
Could it be interested to move this method as a method of `LfBranch` ?

Reply:
Done